### PR TITLE
[TC-763] BE: Login flow

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0763](https://bcdevex.atlassian.net/browse/TC-0763)

Objective:

- BE Changes for supervisor/member login

### CODE REVIEW

File: backend/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    );
+    const member = await this.authService.getLoggedInUser(req);
+    const supervisor = await this.authService.verifySupervisor(req);
+    return {
+      role: req?.role,
+      program: req?.program,
+      username: `${req?.username}`,
+      idir: req?.idir,
+      member: member !== null,
+      supervisor: supervisor,
```
Comment (chelsea-EYDS) [inline backend/src/auth/auth.controller.ts:48]: Might make sense to use an array for Roles but will require more extensive changes. Using role for setting permissions elsewhere, but on this endpoint I am returning member/supervisor boolean in order to know which routes should be available in the user navigation menu.

File: backend/src/auth/auth.module.ts [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+import { TokenGuard } from './token.guard';
 import { LoggerModule } from '../logger/logger.module';
+import { PersonnelModule } from '../personnel/personnel.module';
 
 @Module({
-  imports: [LoggerModule],
-  providers: [AuthGuard, RolesGuard],
+  imports: [LoggerModule, PersonnelModule],
+  providers: [
+    AuthService,
```
Comment (chelsea-EYDS) [inline backend/src/auth/auth.module.ts:14]: Moved these from main into the module in order to avoid explicitly passing in the reflector.

File: backend/src/personnel/ro/personnel.ro.ts [hunk 3]
```diff
@@ -119,12 +118,13 @@ export class PersonnelRO {
...
   supervisorEmail: string;
 
-  @ApiProperty({
-    description: 'Travel preference',
-    required: true,
-    example: TravelPreference.WILLING_TO_TRAVEL_REGION,
-  })
-  travelPreference: TravelPreference;
+  // TODO - confirm if this should be here - I believe this is on the emcr/bcws data? It was throwing an error
+  // @ApiProperty({
```
Comment (chelsea-EYDS) [inline backend/src/personnel/ro/personnel.ro.ts:122]: @fw-noel can you let me know if this should be here? It was throwing an error and it looks like we have this set on  the bcws/emcr side?

Comment (fw-noel) [reply backend/src/personnel/ro/personnel.ro.ts:122]: Yes, this needs to be here for now. They had wanted to add people's preference for travel preference. Not sure why it's throwing an error

File: backend/src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    } else {
+      request['username'] = '';
+    }
+    if (payload.email) {
+      request['idir'] = payload.email;
+    } else {
+      request['idir'] = '';
+    }
+
+    const isMember = await this.personnelService.verifyMember(payload.email);
```
Comment (fw-noel) [inline backend/src/auth/auth.service.ts:37]: Can shorten this to a single request, get all personnel with email or supervisor as this email. Less db requests the better IMO.

File: backend/src/personnel/ro/personnel.ro.ts (hunk 3 above)
Comment (chelsea-EYDS) [reply backend/src/personnel/ro/personnel.ro.ts:122]: okee adding back now sorry!

File: backend/src/auth/auth.service.ts (hunk 4 above)
Comment (chelsea-EYDS) [reply backend/src/auth/auth.service.ts:37]: I agree with this in general, but I think it is more readable this way?

Comment (chelsea-EYDS) [reply backend/src/auth/auth.service.ts:37]: Otherwise we're querying where email === email or where email === supervisor email and then having to figure out whether it was the email that matched or the supervisor email that matched. I'm just trying to keep the role setting as clear as possible since it's part of the auth guard. But open to discussion!

File: backend/src/auth/auth.controller.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    );
+
+    const { isMember, isSupervisor } =
+      await this.authService.verifyMemberOrSupervisor(req.idir);
+
+    return {
+      role: req?.role,
+      program: req?.program,
+      username: `${req?.username}`,
+      idir: req?.idir,
```
Comment (fw-noel) [inline backend/src/auth/auth.controller.ts:48]: At least with local, username is returning the full name and idir is returning the email. Worth a check if this will be the case in SSO.

File: backend/src/auth/auth.controller.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  @ApiOperation({
+    summary: 'Verify the role and sent user data',
+    description: 'Returns the user data with the request',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+  })
+  @Get()
+  async checkUserPermissions(
```
Comment (fw-noel) [inline backend/src/auth/auth.controller.ts:34]: I'd rename, at this point, we're querying user information. So getUserInfo? Or something

**Review by fw-noel — APPROVED** [review]
Minor comments!

File: backend/src/auth/auth.controller.ts (hunk 6 above)
Comment (chelsea-EYDS) [reply backend/src/auth/auth.controller.ts:34]: copying to address in subsequent pr
